In [62]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import operator

In [63]:
df = pd.read_csv('./csvfile/data12.csv')

In [64]:
df.head()

Unnamed: 0  CP ID  Connector  Start Date Start Time    End Date End Time  \
0           0  50230          1  01/04/2018      12:29  01/04/2018    18:52   
1           1  50230          2  01/04/2018      12:11  01/04/2018    12:34   
2           2  50230          2  01/04/2018      12:26  01/04/2018    12:47   
3           3  50230          2  01/04/2018      12:00  01/04/2018    12:16   
4           4  50230          2  01/04/2018      12:09  01/04/2018    13:10   

   Total kWh  
0      12.47  
1       5.54  
2       4.60  
3       3.15  
4      23.40

In [65]:
df.drop(['Unnamed: 0'], axis=1, inplace=True)

In [66]:
df.head()

CP ID  Connector  Start Date Start Time    End Date End Time  Total kWh
0  50230          1  01/04/2018      12:29  01/04/2018    18:52      12.47
1  50230          2  01/04/2018      12:11  01/04/2018    12:34       5.54
2  50230          2  01/04/2018      12:26  01/04/2018    12:47       4.60
3  50230          2  01/04/2018      12:00  01/04/2018    12:16       3.15
4  50230          2  01/04/2018      12:09  01/04/2018    13:10      23.40

In [67]:
# 시간 구하기 위해 시간,분 나눠서 저장
starthour = []
endhour = []
startmin = []
endmin = []
for i in df['Start Time']:
    starthour.append(float(i.split(':')[0]))
    startmin.append(float(i.split(':')[1]))
    

for j in df['End Time']:
    endhour.append(float(j.split(':')[0]))
    endmin.append(float(j.split(':')[1]))



In [57]:
# 연, 월, 일 비교 : 충전시간을 분단위로 구하기 위함

startday = []
endday = []
startmonth = []
endmonth = []
startyear = []
endyear = []

for i in df['Start Date']:
    startday.append(float(i.split('/')[0]))
    startmonth.append(float(i.split('/')[1]))
    startyear.append(float(i.split('/')[2]))
    
for j in df['End Date']:
    startday.append(float(j.split('/')[0]))
    startmonth.append(float(j.split('/')[1]))
    startyear.append(float(j.split('/')[2]))
    
    

In [75]:
subtractHour = []

for i in range(len(starthour)):
    subtractHour.append(endhour[i]-starthour[i])

In [78]:
from collections import Counter
Counter(subtractHour)

Counter({6.0: 45,
         0.0: 437,
         1.0: 477,
         5.0: 38,
         2.0: 166,
         -3.0: 36,
         -5.0: 30,
         -4.0: 27,
         3.0: 80,
         4.0: 59,
         -2.0: 23,
         7.0: 8,
         -6.0: 17,
         9.0: 5,
         -1.0: 3,
         11.0: 2,
         -8.0: 1,
         8.0: 2,
         10.0: 4,
         -9.0: 1})

## Top 3
- 12시에 시작하여 12시 59분 사이 충전 하는 횟수 : 437
- 12시에 시작하여 최대 13시 59분 사이 충전하는 횟수 : 477
- 12시에 시작하여 최대 14시 59분 사이 충전하는 횟수 : 166
- 위 데이터의 음수는 12시에 시작하여 하루를 넘겨서 충전하는 횟수 : 138

In [118]:
#12시에 시작하여 12시 59분사이 까지 충전할 때의 평균 전력량을 구하기 위해 인덱스 추출
k = -1
kwh_12to12 = 0
kwh_12to1 = 0
kwh_12to2 = 0
kwh_12to3 = 0
kwh_12to4 = 0

count = 0
countt = 0
counttt = 0
countttt = 0
counttttt = 0

for i in range(len(starthour)):
    k = k+1
    if(endhour[i]-starthour[i] == 0):
        kwh_12to12 += float(df.loc[k,['Total kWh']])
        count += 1
        

    if(endhour[i]-starthour[i] == 1):
        kwh_12to1 += float(df.loc[k,['Total kWh']])
        countt += 1
        
    if(endhour[i]-starthour[i] == 2):
        kwh_12to2 += float(df.loc[k,['Total kWh']])
        counttt += 1
        
    if(endhour[i]-starthour[i] == 3):
        kwh_12to3 += float(df.loc[k,['Total kWh']])
        countttt += 1
        
    if(endhour[i]-starthour[i] == 4):
        kwh_12to4 += float(df.loc[k,['Total kWh']])
        counttttt += 1
        

kwh_12to12 = kwh_12to12 / count
kwh_12to1 = kwh_12to1 / countt
kwh_12to2 = kwh_12to2 / counttt
kwh_12to3 = kwh_12to3 / countttt
kwh_12to4 = kwh_12to4 / counttttt

print(kwh_12to12)
print(kwh_12to1)
print(kwh_12to2)
print(kwh_12to3)
print(kwh_12to4)

6.655475972540044
8.065582809224324
5.9660060240963855
8.725237500000002
7.3375593220339015


## Top 5
- 12시에 시작하여 12시 59분 사이 충전 할때의 평균 전력량 : 6.655475972540044
- 12시에 시작하여 최대 13시 59분 사이 충전 할때의 평균 전력량 : 8.065582809224324
- 12시에 시작하여 최대 14시 59분 사이 충전 할때의 평균 전력량 : 5.9660060240963855
- 12시에 시작하여 최대 15시 59분 사이 충전 할때의 평균 전력량 : 8.725237500000002
- 12시에 시작하여 최대 16시 59분 사이 충전 할때의 평균 전력량 : 7.3375593220339015
